vcf2hl7v2/common.py

```diff
@@ -0,0 +1,163 @@
+        elif len(alleles) >= 2:
+            allelic_state = 'homozygous'
+            allelic_code = 'LA6705-3'
+        elif sample.gt_type is not None and len(alleles) == 1:
+            allelic_state = 'hemizygous'
+            allelic_code = 'LA6707-9'
+        else:
+            _error_log_allelicstate(record)
+    elif (
+            sample.gt_type is not None and
+            len(alleles) == 1 and
+            alleles[0] == '1'):
+        if hasattr(sample.data, 'AD') and hasattr(sample.data, 'DP'):
+            try:
+                if(
+                        isinstance(sample.data.AD, list) and
+                        len(sample.data.AD) > 0):
+                    ratio = float(
+                        sample.data.AD[0]) / float(sample.data.DP)
+                else:
+                    ratio = float(sample.data.AD) / float(sample.data.DP)
+                if ratio > ratio_ad_dp:
+                    allelic_state = "homoplasmic"
+                    allelic_code = "LA6704-6"
+                else:
+                    allelic_state = "heteroplasmic"
+                    allelic_code = "LA6703-8"
+            except Exception as e:
+                general_logger.debug(e)
+                _error_log_allelicstate(record)
+                pass
+        else:
+            _error_log_allelicstate(record)
+    else:
+        _error_log_allelicstate(record)
+    return {'ALLELE': allelic_state, 'CODE': allelic_code}
+
+
+def extract_chrom_identifier(chrom):
+    chrom = chrom.upper().replace("CHR", "")
+    if chrom == "MT":
+        chrom = "M"
+    return chrom
+
+
+def validate_chrom_identifier(chrom):
+    chrom = extract_chrom_identifier(chrom)
+    pattern = '^[1-9]$|^1[0-9]$|^2[0-2]$|^[XYM]$'
+    result = re.match(pattern, chrom)
+    return bool(result)
+
+
+def validate_ratio_ad_dp(ratio_ad_dp):
+    if not (ratio_ad_dp):
+        return False
+    if not isinstance(ratio_ad_dp, float):
+        return False
+    if ratio_ad_dp < 0 or ratio_ad_dp >= 1:
+        return False
+    return True
+
+
+def validate_has_tabix(has_tabix):
+    if not isinstance(has_tabix, bool):
+        return False
+    return True
+
+
+def validate_seed(seed):
+    if isinstance(seed, int) and not isinstance(seed, bool):
+        return True
+    return False
+
+
+def get_unique_identifier(n):
+
+    def div(n):
+        if ((n % 26) == 0 and n > 26):
+            return get_unique_identifier(n//26 - 1)
+        elif (n//26) <= 0:
+            return ''
+        elif n == 26:
+            return 'z'
+        elif (n//26) in D:
+            return D[n//26]
+        else:
+            return get_unique_identifier(n//26)
+
+    def mod(n):
+        if n <= 0:
+            return ''
+        elif ((n % 26) == 0 and n > 26):
+            return 'z'
+        elif (n % 26) <= 0:
+            return ''
+        else:
+            return D[n % 26]
+
+    res_div = div(n)
+    res_mod = mod(n)
+    return (res_div + res_mod)
```

Comment:
this can be simplified:

```
n = 1234;

def alphabet_index(n):
    ans = ""
    while(n != 0):
        remainder = n % 26;
        n = n//26;
        if remainder == 0:
            remainder = 26
            n = n-1
        ans = chr(96 + remainder) + ans
    return ans




def test_alphabet_index():
    index = [1, 26, 29, 79, 207, 703, 1234, 52]
    expected = ['a', 'z', 'ac', 'ca', 'gy', 'aaa', 'aul', 'az']
    i = 0
    for value in index:
        actual = alphabet_index(value)
        print(actual)
        if actual == expected[i]:
            print("true")
        else:
            print("false")
        i += 1
        
test_alphabet_index()
```